src/lib/type_cast.hpp

```diff
@@ -33,20 +33,33 @@ const T& get(const AllTypeVariant& value) {
   return boost::get<T>(value);
 }
 
-// cast methods - from variant to specific type
+// cast methods - from one type to another
 
+// For convenience, allow `type_cast<T>(T bla);`, but it shouldn't do anything.
+// Conversions that don't require lexical_cast handling (e.g., float->double or float->int) are handled here as well.
+template <typename T, typename U, typename = std::enable_if_t<std::is_convertible_v<std::decay<U>, std::decay<T>>>>
+inline __attribute__((always_inline)) auto&& type_cast(U&& value) {
```

Comment:
Time for the `HYRISE_FORCE_INLINE` macro? ;)

Reply:
What would that achieve?

Reply:
Sorry, just a side remark - every other codebase seems to have such a macro since `__attribute__((always_inline))` looks just sooo ugly.

Reply:
I don't like the syntax either, but would still prefer it over what I consider to be a useless indirection